content/en/blog/2025/expose-otel-collector-gateway-api/index.md

```diff
@@ -0,0 +1,621 @@
+[v1.2](https://gateway-api.sigs.k8s.io/implementations/v1.2/). Install the
+Gateway API CRDs if not present:
+
+```bash
+kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null || \
+kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.2.1/standard-install.yaml
+
+#To support GRPCRoute as of now. Would not be required once GRPCRoute CRD becomes GA.
+kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v1.2.0" | kubectl apply -f -
+```
+
+### Step 2: Generate Self Signed Certificates
+
+In order to setup mTLS between the client and the server, we need a set of
+certificates. For this demo scenario, we'll be using self-signed certificate. We
+will be using the same CA for signing both clients and server in this demo.
+We'll use `openssl` to create our certificates. Please refer to the `openssl`
+documentation for configuration details.
+
+```bash
+# Variables (adjust domain/names as needed)
+export ROOT_CA_SUBJ="/CN=MyDemoRootCA"
+# Use a relevant CN/SAN for the server/gateway. If clients connect via IP, include it.
+# For DNS, use the hostname clients will use (e.g., otel.example.com)
+export SERVER_HOSTNAME="otel-gateway.example.com"
+export SERVER_SUBJ="/CN=${SERVER_HOSTNAME}"
+export CLIENT_SUBJ="/CN=external-otel-client"
+
+# 1. Create Root CA certificate and key
+openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj "${ROOT_CA_SUBJ}" -keyout rootCA.key -out rootCA.crt
+
+# 2. Create Server CSR & sign with Root CA
+openssl req -newkey rsa:4096 -nodes -keyout server.key -out server.csr -subj "${SERVER_SUBJ}" \
+  -addext "subjectAltName = DNS:${SERVER_HOSTNAME}" # Add SAN for hostname validation
+
+openssl x509 -req -in server.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out server.crt -days 365 -sha256 \
+  -extfile <(printf "subjectAltName=DNS:${SERVER_HOSTNAME}") # Ensure SAN is in the final cert
+
+# 3. Create Client CSR and sign with Root CA
+openssl req -newkey rsa:4096 -nodes -keyout client.key -out client.csr -subj "${CLIENT_SUBJ}"
+
+openssl x509 -req -in client.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out client.crt -days 365 -sha256
+```
+
+{{% alert %}}
+
+For production, never use self-signed certificates for external-facing endpoints
+accessible from the public internet. Use certificates issued by a trusted public
+CA (e.g., Let's Encrypt through cert-manager) or a managed internal PKI system.
+The process of obtaining certs would differ, but the concepts of using them in
+Kubernetes remain similar. Ensure the server certificate's Common Name (CN) or
+Subject Alternative Name (SAN) matches the hostname clients use to connect.
+
+{{% /alert %}}
+
+## Step 3: Create `otel-collector` namespace
+
+We will deploy our OTel Collector setup in the given namespace. Further on,
+depending on the Gateway/service mesh implementation you use, you may configure
+the namespace accordingly. For e.g: with Istio, we can the namespace with
```

Comment:
Seems like there's a word missing from this sentence. I added "create" but that might not be correct. Also better to spell out "example" in this case.

```suggestion
the namespace accordingly. For example, with Istio, we can create the namespace with
```